bundles/CoreBundle/src/EventListener/Frontend/ElementListener.php

```diff
@@ -17,6 +17,7 @@
 
 use Pimcore\Bundle\AdminBundle\Security\User\UserLoader;
 use Pimcore\Bundle\CoreBundle\EventListener\Traits\PimcoreContextAwareTrait;
+use Pimcore\Bundle\StaticRoutesBundle\Model\Staticroute;
```

Comment:
how does it work when the bundle is disabled?

Reply:
Maybe we can talk about this, because I am not sure how to refactor this method
```php
  protected function applyTargetGroups(Request $request, Document $document)
    {
        if (!$document instanceof Document\Targeting\TargetingDocumentInterface || null !== Staticroute::getCurrentRoute()) {
            return;
        }

        // reset because of preview and editmode (saved in session)
        $document->setUseTargetGroup(null);

        $this->targetingConfigurator->configureTargetGroup($document);

        if ($document->getUseTargetGroup()) {
            $this->logger->info('Setting target group to {targetGroup} for document {document}', [
                'targetGroup' => $document->getUseTargetGroup(),
                'document' => $document->getFullPath(),
            ]);
        }
    }
```

Reply:
I think here we need to check whether the targeting bundle is installed or not, maybe just by using a `class_exists()`.